kiwmi/desktop/output.c

```diff
@@ -376,8 +388,143 @@ new_output_notify(struct wl_listener *listener, void *data)
+        wl_container_of(listener, desktop, output_layout_change);
+    struct wlr_output_configuration_v1 *configuration =
+        wlr_output_configuration_v1_create();
+
+    struct kiwmi_output *output;
+    wl_list_for_each (output, &desktop->outputs, link) {
+        struct wlr_output_configuration_head_v1 *head =
+            wlr_output_configuration_head_v1_create(
+                configuration, output->wlr_output);
+
+        struct wlr_box *area = wlr_output_layout_get_box(
+            desktop->output_layout, output->wlr_output);
+
+        if (area != NULL) {
+            head->state.x = area->x;
+            head->state.y = area->y;
+        }
+    }
+
+    wlr_output_manager_v1_set_configuration(
+        desktop->output_manager, configuration);
+}
+
+static void
+output_manager_configure(
+    struct kiwmi_server *server,
+    struct wlr_output_configuration_v1 *configuration,
+    bool test)
+{
+    struct wlr_output_configuration_head_v1 *head;
+    struct wlr_output_configuration_head_v1 *bad_head = NULL;
+    wl_list_for_each (head, &configuration->heads, link) {
+        struct wlr_output *wlr_output = head->state.output;
+
+        wlr_output_enable(wlr_output, head->state.enabled);
+        if (head->state.enabled) {
+            if (head->state.mode) {
+                wlr_output_set_mode(wlr_output, head->state.mode);
+            } else {
+                wlr_output_set_custom_mode(
+                    wlr_output,
+                    head->state.custom_mode.width,
+                    head->state.custom_mode.height,
+                    head->state.custom_mode.refresh);
+            }
+
+            wlr_output_set_transform(wlr_output, head->state.transform);
+            wlr_output_set_scale(wlr_output, head->state.scale);
+        }
+
+        if (!wlr_output_test(wlr_output)) {
+            bad_head = head;
+            break;
+        }
+    }
+
+    wl_list_for_each (head, &configuration->heads, link) {
+        if (bad_head == NULL && !test) {
+            struct kiwmi_output *output = head->state.output->data;
+            bool enable_changed         = WLR_OUTPUT_STATE_ENABLED
+                == (output->wlr_output->pending.committed
+                    & WLR_OUTPUT_STATE_ENABLED);
+
+            if (head->state.enabled) {
+                if (enable_changed) {
+                    wlr_output_layout_add(
+                        server->desktop.output_layout,
+                        output->wlr_output,
+                        head->state.x,
+                        head->state.y);
+                    wl_signal_emit(&output->desktop->events.new_output, output);
+                } else {
+                    wlr_output_layout_move(
+                        server->desktop.output_layout,
+                        output->wlr_output,
+                        head->state.x,
+                        head->state.y);
+                    wl_signal_emit(&output->events.resize, output);
```

Comment:
This event is emitted even if the output isn’t resized. I think a check similar to `enable_changed` would fix that. (Or should it be preserved so that all outputs emit some event?)

Reply:
The resize signal is now emitted if there are pending mode, scale or transform changes, or if head->state.x/y are different to the current values.  Having moves trigger a resize event is a bit weird but I think the config should be told when an output moves and having seperate move and resize events doesn't seem worth the effort.